client.go

```diff
@@ -67,7 +67,44 @@ func WithOnClose(onClose func()) ClientOpts {
 // WithUnaryClientInterceptor sets the provided client interceptor
 func WithUnaryClientInterceptor(i UnaryClientInterceptor) ClientOpts {
 	return func(c *Client) {
-		c.interceptor = i
+		if c.interceptor == nil {
```

Comment:
IMO, we should not update this function because the signature is to assign the interceptor instead of adding. If the users want to use multiple interceptors, they should use `WithChainUnaryClientInterceptor`.

Reply:
Yes, true. I updated it like that to simply set the interceptor to the given one. Only `WithChainUnaryClientInterceptor` now chains interceptors.